deps/verifier/src/snp/mod.rs

```diff
@@ -44,57 +49,93 @@ const UCODE_SPL_OID: Oid<'static> = oid!(1.3.6 .1 .4 .1 .3704 .1 .3 .8);
 const SNP_SPL_OID: Oid<'static> = oid!(1.3.6 .1 .4 .1 .3704 .1 .3 .3);
 const TEE_SPL_OID: Oid<'static> = oid!(1.3.6 .1 .4 .1 .3704 .1 .3 .2);
 const LOADER_SPL_OID: Oid<'static> = oid!(1.3.6 .1 .4 .1 .3704 .1 .3 .1);
+const FMC_SPL_OID: Oid<'static> = oid!(1.3.6 .1 .4 .1 .3704 .1 .3 .9);
 
 // KDS URL parameters
 const KDS_CERT_SITE: &str = "https://kdsintf.amd.com";
 const KDS_VCEK: &str = "/vcek/v1";
 
 /// Attestation report versions supported
-const REPORT_VERSION_MIN: u32 = 2;
-const REPORT_VERSION_MAX: u32 = 3;
+const REPORT_VERSION_MIN: u32 = 3;
+const REPORT_VERSION_MAX: u32 = 4;
 
-#[derive(Debug)]
+#[derive(Default, Debug)]
 pub struct Snp {
-    vendor_certs: VendorCertificates,
-}
-
-/// Loads the Milan certificate chain and returns a static reference to it.
-/// The chain is loaded lazily using `OnceLock` to ensure it's only initialized once.
-/// Certificates are loaded from a PEM file and must contain exactly three certificates (ASK, ARK, ASVK).
-pub(crate) fn load_milan_cert_chain() -> &'static Result<VendorCertificates> {
-    static MILAN_CERT_CHAIN: OnceLock<Result<VendorCertificates>> = OnceLock::new();
-    MILAN_CERT_CHAIN.get_or_init(|| {
-        let certs = X509::stack_from_pem(include_bytes!("milan_ask_ark_asvk.pem"))?;
-        if certs.len() != 3 {
-            bail!("Malformed Milan ASK/ARK/ASVK");
-        }
-
-        let vendor_certs = VendorCertificates {
-            ask: certs[0].clone(),
-            ark: certs[1].clone(),
-            asvk: certs[2].clone(),
-        };
-        Ok(vendor_certs)
-    })
+    vendor_certs: HashMap<ProcType, VendorCertificates>,
 }
 
 impl Snp {
-    /// Creates a new `Snp` instance by loading the Milan certificate chain.
-    /// Returns an error if the certificate chain can not be loaded.
+    /// Creates a new `Snp` instance by loading the SNP certificate chain into a hash-map.
+    /// Returns an error if the certificate chains can not be loaded.
     pub fn new() -> Result<Self> {
-        let Result::Ok(vendor_certs) = load_milan_cert_chain() else {
-            bail!("Failed to load Milan cert chain");
+        let Result::Ok(vendor_certs) = load_cert_chain() else {
+            bail!("Failed to load stored SNP vendor certs");
         };
         let vendor_certs = vendor_certs.clone();
```

Comment:
I noticed that the `load_cert_chain()` function is called only once but will be `clone()` for every `Snp::new()`. This implies cert data memory copy every time a verification request comes but the data of certs is static and constant. 

Another way to design this is use a `static LazyLock` on top of the module to include the HashMap<HashMap<ProcType, VendorCertificates>> of the module and get rid of the `vendor_certs` member of `Snp`. Then try to directly reference the static variable wherever needs.

This is relative easy like
```rust
static CERT_CHAINS: LazyLock<HashMap<ProcType, VendorCertificates>> = LazyLock::new(|| {
    ....// initial the map and feel free to panic or unwrap as the cert files are statically included thus
    // will not cause unexpected errors
}
```